tests/components/home_connect/test_init.py

```diff
@@ -156,72 +144,61 @@ async def test_api_setup(
-
-    # Second re-load is blocked by Throttle.
-    assert await hass.config_entries.async_unload(config_entry.entry_id)
-    assert config_entry.state == ConfigEntryState.NOT_LOADED
-    freezer.tick(SCAN_INTERVAL.seconds - 0.1)
-    assert await hass.config_entries.async_setup(config_entry.entry_id)
-    assert get_appliances.call_count == get_appliances_call_count + 1
-
-
-@pytest.mark.usefixtures("bypass_throttle")
 async def test_exception_handling(
-    integration_setup: Callable[[], Awaitable[bool]],
+    integration_setup: Callable[[MagicMock], Awaitable[bool]],
     config_entry: MockConfigEntry,
     setup_credentials: None,
-    get_appliances: MagicMock,
-    problematic_appliance: Mock,
+    client_with_exception: MagicMock,
 ) -> None:
     """Test exception handling."""
-    get_appliances.return_value = [problematic_appliance]
     assert config_entry.state == ConfigEntryState.NOT_LOADED
-    assert await integration_setup()
+    assert await integration_setup(client_with_exception)
     assert config_entry.state == ConfigEntryState.LOADED
 
 
 @pytest.mark.parametrize("token_expiration_time", [12345])
-@pytest.mark.usefixtures("bypass_throttle")
+@respx.mock
 async def test_token_refresh_success(
-    integration_setup: Callable[[], Awaitable[bool]],
+    hass: HomeAssistant,
+    platforms: list[Platform],
+    integration_setup: Callable[[MagicMock], Awaitable[bool]],
     config_entry: MockConfigEntry,
     aioclient_mock: AiohttpClientMocker,
     requests_mock: requests_mock.Mocker,
     setup_credentials: None,
+    client: MagicMock,
 ) -> None:
     """Test where token is expired and the refresh attempt succeeds."""
 
     assert config_entry.data["token"]["access_token"] == FAKE_ACCESS_TOKEN
 
     requests_mock.post(OAUTH2_TOKEN, json=SERVER_ACCESS_TOKEN)
-    requests_mock.get("/api/homeappliances", json={"data": {"homeappliances": []}})
-
     aioclient_mock.post(
         OAUTH2_TOKEN,
         json=SERVER_ACCESS_TOKEN,
     )
-    assert await integration_setup()
+    appliances = client.get_home_appliances.return_value
+
+    async def mock_get_home_appliances():
+        await client._auth.async_get_access_token()
+        return appliances
+
+    client.get_home_appliances.return_value = None
+    client.get_home_appliances.side_effect = mock_get_home_appliances
+
+    def init_side_effect(auth) -> MagicMock:
+        client._auth = auth
+        return client
+
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    with (
+        patch("homeassistant.components.home_connect.PLATFORMS", platforms),
+        patch("homeassistant.components.home_connect.HomeConnectClient") as client_mock,
+    ):
+        client_mock.side_effect = MagicMock(side_effect=init_side_effect)
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+        await hass.async_block_till_done()
+    config_entry.runtime_data.client = client
```

Comment:
This shouldn't be needed if we patch correctly. We shouldn't touch or use `runtime_data` in the tests.
```suggestion
```